feat(ffe-form-react): Add support for forwarding refs

selbekk opened the pull request:
BREAKING CHANGE: Requires react@^16.3

This commit adds support for forwarding refs to the `Input` and
`TextArea` components, so consumers can handle focus programmatically.

---

Since this is a bump in peer dependencies, this is per definition a breaking change - but it's a small and simple one to deal with for most consumers.

Fixes #231

selbekk commented:
Hm, looks like some of the tests break (but only when running jest in ci-mode). I'm gonna look into it. :)

selbekk commented:
Related to this: https://github.com/facebook/jest/issues/5441

ivarni commented:
Could you perhaps add some documentation and an example?

selbekk commented:
> Could you perhaps add some documentation and an example?

In my mind, this is something that should "just work" out of the box - but I don't mind adding an example

ivarni commented:
Well, yeah, but since we're not doing this for all our components it might be worth pointing out to someone that isn't familiar with the codebase that we're doing it for this one?

Commit:
feat(ffe-form-react): Add support for forwarding refs

BREAKING CHANGE: Requires react@^16.3

This commit adds support for forwarding refs to the `Input` and
`TextArea` components, so consumers can handle focus programmatically.

Commit:
docs(ffe-form-react): Document how to control focus with refs

selbekk commented:
There - similar examples for both textarea and input components are added - and I think I fixed the tests as well. #famouslastwords

ivarni commented on packages/ffe-form-react/src/Input.md:
```diff
@@ -6,7 +6,7 @@ Vi har tre varianter av inputfelt:
 3. text-like
 
-_Standard_ og _inline_ er mest relevant for utviklere. De ser helt like ut, men opprører seg annerledes i samspillet
+_Standard_ og _inline_ er mest relevant for utviklere. De ser helt like ut, men oppfører seg annerledes i samspillet
```
keep funny spelling error or we riot!

selbekk commented:
Do you want to merge this at some point @antidecaf or @kwltrs ? Since it's a breaking change (at least on paper), I'd like an OK from at least one of you.

antidecaf reviewed (approved):
LGTM
